### PR TITLE
cilium: fix ipv6 neighbor discovery

### PR DESCRIPTION
**Pull request by borkmann**
It was never working in KPR due to the historic use of the ARPing library. Now that we moved to managed neighbors in https://github.com/cilium/cilium/pull/17713, we can make KPR working with backends in same L2 and piggy-back on kernel for discovery.

### CODE REVIEW

**Comment by borkmann** [comment]
test-runtime

**Comment by borkmann** [comment]
test-runtime

**Comment by borkmann** [comment]
test-runtime

**Comment by borkmann** [comment]
Travis hitting:
```
docker rm -f "cilium-consul-test-container" 2> /dev/null
rm -rf /tmp/cilium-consul-certs
mkdir /tmp/cilium-consul-certs
cp /home/travis/gopath/src/github.com/cilium/cilium/test/consul/* /tmp/cilium-consul-certs
chmod -R a+rX /tmp/cilium-consul-certs
docker run -d \
	--name "cilium-consul-test-container" \
	-p 8501:8443 \
	-e 'CONSUL_LOCAL_CONFIG={"skip_leave_on_interrupt": true, "disable_update_check": true}' \
	-v /tmp/cilium-consul-certs:/cilium-consul/ \
	consul:1.7.2 \
	agent -client=0.0.0.0 -server -bootstrap-expect 1 -config-file=/cilium-consul/consul-config.json
Unable to find image 'consul:1.7.2' locally
1.7.2: Pulling from library/consul
docker: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit.
See 'docker run --help'.
make: *** [Makefile:155: start-kvstores] Error 125
```

**Comment by borkmann** [comment]
test-runtime

**Comment by borkmann** [comment]
test-runtime

**Comment by borkmann** [comment]
CI issue in runtime test:
```
01:31:33  Checking for older boxes...
01:31:33  Removing box 'cilium/ubuntu' (v208) with provider 'virtualbox'...
01:31:33  Removing box 'cilium/ubuntu-4-19' (v48) with provider 'virtualbox'...
01:31:33  Removing box 'cilium/ubuntu-5-4' (v12) with provider 'virtualbox'...
01:31:33  Removing box 'cilium/ubuntu-next' (v105) with provider 'virtualbox'...
01:31:33  VBoxManage: error: Machine '955bc943-6d63-4970-8c85-cc3f82882ba3' is not currently running
01:31:34  0%...10%...20%...30%...40%...
01:31:34  Progress state: VBOX_E_OBJECT_IN_USE
01:31:34  VBoxManage: error: Machine delete failed
01:31:34  VBoxManage: error: Cannot close medium '/root/VirtualBox VMs/ubuntu-18.04.6-amd64-105_1636588893675_83151/ubuntu-18.04.6-amd64-105-disk001.vmdk' because it has 1 child media
01:31:34  VBoxManage: error: Details: code VBOX_E_OBJECT_IN_USE (0x80bb000c), component MediumWrap, interface IMedium
01:31:34  VBoxManage: error: Context: "RTEXITCODE handleUnregisterVM(HandlerArg*)" at line 162 of file VBoxManageMisc.cpp
01:31:34  VBoxManage: error: Machine 'f05120e2-206e-4863-9229-6e38ce297658' is not currently running
01:31:35  0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
```

**Comment by borkmann** [comment]
test-runtime

**Comment by borkmann** [comment]
test-runtime

**Comment by borkmann** [comment]
test-runtime

File: pkg/datapath/linux/node.go [hunk 1]
```diff
@@ -1449,7 +1449,7 @@ func (n *linuxNodeHandler) NodeConfigurationChanged(newConfig datapath.LocalNode
 	prevConfig := n.nodeConfig
 	n.nodeConfig = newConfig
 
-	if n.nodeConfig.EnableIPv4 {
+	if n.nodeConfig.EnableIPv4 || n.nodeConfig.EnableIPv6 {
```
Comment (jrfastab) [inline pkg/datapath/linux/node.go:1452]: Is there even a case where both ipv4 and ipv6 are disabled? Seems we can always have this on and skip check?

**Review by jrfastab — APPROVED** [review]
Small point at least the ipv4 fix should likely be backported. PR looks good though.

File: pkg/datapath/linux/node.go (hunk 1 above)
Comment (borkmann) [reply pkg/datapath/linux/node.go:1452]: I think not, I can remove the check in a follow-up, liked the explicit check, but either way is fine.

**Comment by borkmann** [comment]
> Small point at least the ipv4 fix should likely be backported. PR looks good though.

Yes, the nextHop refcount fix would need to be backported separately.

**Comment by borkmann** [comment]
Merging as CI is green, acked by John, and tests passed in order to make it for rc1. @brb lets sync tomorrow if there's anything else that needs to be followed-up. Meanwhile will ping reporter in #17693 for testing latest.

**Pull request merged**